File: src/ProjectR.Scripting/content/scripts/spells/NightStalker.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Night Stalker"
end

function GetDescription()
  return "Attack all enemies with shadowy power."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .88
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  local damage = (4 * aAD * (aDRK/100) - dDEF) * (100/dDRK)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/NightStalker.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class NightStalker : SpellScriptBase
    {
        public override string Name { get { return "Night Stalker"; } }
        public override string Description { get { return "Attack all enemies with shadowy power."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4 * cAD * (cDRK / 100) - tDEF) * (100 / tDRK);
            DealDamage(damage);
        }
    }
}